Class/Warrior/Spells/SunderArmor.cpp

```diff
@@ -0,0 +1,81 @@
+#include "SunderArmor.h"
+
+#include "Buff.h"
+#include "CastComplete.h"
+#include "CharacterStats.h"
+#include "CombatRoll.h"
+#include "CooldownControl.h"
+#include "HeroicStrike.h"
+#include "MainhandAttack.h"
+#include "OffhandAttack.h"
+#include "Warrior.h"
+#include "WarriorSpells.h"
+
+SunderArmor::SunderArmor(Warrior* pchar, WarriorSpells* spells) :
+    // TODO Find the right spell icon
+    Spell("SunderArmor", "Assets/ability/Ability_warrior_decisivestrike.png",
+          pchar, new CooldownControl(pchar, 0.0), RestrictedByGcd::Yes,
+          ResourceType::Rage,
+          15),
+    TalentRequirer(QVector<TalentRequirerInfo*>{new TalentRequirerInfo("Improved Sunder Armor", 3, DisabledAtZero::No)}),
+    warr(pchar),
+    spells(spells),
+    // TODO find these lower level values
+    innate_threat_by_spell_rank({0, 0, 0, 0, 260 }),
+    innate_threat(260),
+    talent_ranks({15, 14, 13, 12})
+{
+}
+
+SunderArmor::~SunderArmor() {
+    delete cooldown;
+}
+
+void SunderArmor::increase_talent_rank_effect(const QString&, const int curr) {
+    resource_cost = talent_ranks[curr];
+}
+
+void SunderArmor::decrease_talent_rank_effect(const QString&, const int curr) {
+    resource_cost = talent_ranks[curr];
+}
+
+SpellStatus SunderArmor::is_ready_spell_specific() const {
+    if (warr->in_battle_stance())
+      return SpellStatus::InBattleStance;
+
+    if (warr->in_berserker_stance())
+      return SpellStatus::InBerserkerStance;
+
+    return warr->on_stance_cooldown() ? SpellStatus::OnStanceCooldown : SpellStatus::Available;
+}
+
+void SunderArmor::spell_effect() {
+    // Sunder armor can't crit
+    const int result = roll->get_melee_ability_result(warr->get_mh_wpn_skill(), 0);
+
+    cooldown->add_gcd_event();
+
+    if (result == PhysicalAttackResult::MISS) {
+        increment_miss();
+        warr->lose_rage(static_cast<unsigned>(round(warr->get_resource_level(resource_type) * 0.16)));
+        return;
+    }
+    if (result == PhysicalAttackResult::DODGE) {
+        increment_dodge();
+        spells->get_overpower_buff()->apply_buff();
+        warr->lose_rage(static_cast<unsigned>(round(warr->get_resource_level(resource_type) * 0.16)));
+        return;
+    }
+    if (result == PhysicalAttackResult::PARRY) {
+        increment_parry();
+        warr->lose_rage(static_cast<unsigned>(round(warr->get_resource_level(resource_type) * 0.16)));
+        return;
+    }
+
+    const unsigned rage = warr->get_resource_level(resource_type);
+    warr->lose_rage(rage);
+    if (result == PhysicalAttackResult::HIT) {
```

Comment:
You can remove this `if` and always invoke `warr->melee_mh_yellow_hit_effect`, since the other flows lead to a `return`, and you passed zero crit to the function.

Side note: are we certain SA "crits" does not trigger on crit proc effects? I've heard talk about that a few months ago (and may have been confirmed by now).

Reply:
> Side note: are we certain SA "crits" does not trigger on crit proc effects? I've heard talk about that a few months ago (and may have been confirmed by now).

I do not know the answer to this. Do you have a source?

Reply:
Sorry I do not, it was among guildies at the time.